prometheus/histogram.go

```diff
@@ -1575,3 +1593,152 @@ func addAndResetCounts(hot, cold *histogramCounts) {
 	atomic.AddUint64(&hot.nativeHistogramZeroBucket, atomic.LoadUint64(&cold.nativeHistogramZeroBucket))
 	atomic.StoreUint64(&cold.nativeHistogramZeroBucket, 0)
 }
+
+type nativeExemplars struct {
+	exemplars                       map[*dto.Exemplar]*totalItem
+	nativeHistogramExemplarTTL      time.Duration
+	nativeHistogramMaxExemplarCount uint32
+	timeList                        *timeItem
+	logarithmList                   *logarithmItem
+}
+
+type totalItem struct {
+	timeItem      *timeItem
+	logarithmItem *logarithmItem
+}
+
+func newNativeExemplars(ttl time.Duration, count uint32) *nativeExemplars {
+	return &nativeExemplars{
+		nativeHistogramExemplarTTL:      ttl,
+		nativeHistogramMaxExemplarCount: count,
```

Comment:
These two need reasonable defaults in case they are left at their zero value (something like 5m for the TTL and 10 for the count?). And the defaults need to be documented in the doc comment for the options, too.

Reply:
add default value and comments in doc.